### ISSUE TITLE
Can't get debugging running! (VSC-676)

### ISSUE THREAD

**Issue by mickpf** [espressif/vscode-esp-idf-extension#423]
Hello,

Since several days I'm trying to get either the Eclipse plugin or the VS Code Extension running so that I can debug my code with ESP-IDF. I gave up trying to get the Eclipse plug-in running properly because the project management does not work as it should do it, problems with the toolbar "Launch Target" and some more problems.

Now I'm trying to set up the VS Code Extension. I can't get the debugging runnung!

Could you help me, please?

In the following I have put the problems together.
Please let me know if you need any further data.

1. After setting up a new workspace on the first attempt to flash, you will be asked for the type: UART or JTAG. There is no way to change this type later (except directly with a text editor in the file "${PROJECT_NAME}.code-workspace: <" idf.flashType ":" UART ">). Changing the file can possibly trigger side effects (affect other projects)!

2. When starting VS Code with a workspace, a new entry appears in the log file "${HOME}/.vscode/extensions/espressif.esp-idf-extension-1.0.3/esp_idf_vsc_ext.log":

    {
        "message": "Open On-Chip Debugger  v0.10.0-esp32-20210401 (2021-04-01-15:45)
                    Licensed under GNU GPL v2\nFor bug reports, read
                        http://openocd.org/doc/doxygen/bugs.html",
        "stack": "Error: Open On-Chip Debugger  v0.10.0-esp32-20210401 (2021-04-01-15:45)
                    Licensed under GNU GPL v2
                    For bug reports, read
                        http://openocd.org/doc/doxygen/bugs.html
                        
                        at /home/mick/.vscode/extensions/espressif.esp-idf-extension-1.0.3/dist/extension.js:1:8461
                        at ChildProcess.exithandler (child_process.js:308:7)
                        at ChildProcess.emit (events.js:315:20)
                        at maybeClose (internal/child_process.js:1048:16)
                        at Socket.<anonymous> (internal/child_process.js:439:11)
                        at Socket.emit (events.js:315:20)\n\tat Pipe.<anonymous> (net.js:673:12)",
        "level": "error",
        "timestamp": "2021-06-02T06:41:08.752Z"
    }

3. Pressing the button "Debug" does not lead to any reaction or action (no log entries, no output, nothing!).

    ==> I changed the file "${workspaceFolder}/.vscode/launch.json" :

        {
            "version": "0.2.0",
            "configurations": [
                {
                    "type": "espidf",
                    "name": "Launch",
                    "request": "launch",
                    "debugPort": 9998,
                    "logLevel": 2,
                    "mode": "manual",
                    "initGdbCommands": [
                        "target remote :3333",
                        "symbol-file /home/mick/Projects/ESP32/EVE/build/EVE.elf",
                        "mon reset halt",
                        "flushregs",
                        "thb app-main"
                    ]
                }
            ]
        }    

    This change leads to the output as can be seen in the attachment "3. Output Button Debug 1.png".
    
    ==> Execution of "ESP-IDF: Select OpenOCD Board Configuration", selection "ESP32 chip (via ESP-PROG)"

        The attempt to start "OpenOCD Server" leads to the output as can be seen in the attachment "3. Output Start OpenOCD Server 1.png".

        Error detected: The file name "esp32_devkitjv1.cfg" is wrong! The correct file name is "esp32_devkitj_v1.cfg".

    ==> The execution of "ESP-IDF: Device configuration", selection "OpenOCD config files" displays and saves the correct file name.
    
    Starting "OpenOCD Server" again leads to the output as can be seen in the attachment "3. Output Start OpenOCD Server 2.png". It works now!
    
    Pressing the "Debug" button does not lead to any reaction or action (no log entries, no output, nothing!).
    
    At this point I can't get any further!

4. Project-related settings for ESP-IDF are entered as global under "Preferences: Open Settings (JSON)":
         "idf.customExtraPaths", "idf.customExtraVars", "idf.port" etc.
    That has already led to problems!

5. Can you use variables like "${workspaceFolder}" and "${command: espIdf.getProjectName}" in the file "launch.json", section "initGdbCommands"?

6. The execution of "Build" generates the warning as can be seen in the attachment "6. Compiler Warnings.png".
     I've tried to supress this warning by the folling entry in the file "${PROJECT_ROOT}/CMakeLists.txt"
     
         target_compile_options (mbedtls PRIVATE -Wno-dev)
         
     It has no effects. What is the right way?

I've tried to attach the named screenshots (format: png, size < 120 kB per file), but it does not work!
How can I send them to you?

Kind Regards,
Michael

**Comment by igrr** [comment]
> The execution of "Build" generates the warning as can be seen in the attachment "6. Compiler Warnings.png".
> I've tried to supress this warning by the folling entry in the file "${PROJECT_ROOT}/CMakeLists.txt"
>   target_compile_options (mbedtls PRIVATE -Wno-dev)
> It has no effects. What is the right way?

I can comment on this one — the issue has been reported in https://github.com/espressif/esp-idf/issues/6767. We have fixed the issue internally but the fix hasn't yet been synced to Github. The patch which fixes the issue is attached.

The reason why adding `target_compile_options (mbedtls PRIVATE -Wno-dev)` doesn't work is because the warning is being reported by CMake. However `target_compile_options` adds arguments to be passed to the C compiler, not to CMake.


[13568.patch.zip](https://github.com/espressif/vscode-esp-idf-extension/files/6585912/13568.patch.zip)

**Comment by mickpf** [comment]
Hello Ivan,

I'm sorry, but I don't have neither the time nor inclination to learn python and/or cmake. So I couldn't turn off this annoying warning on my own.
Thank you for your help!

**Comment by brianignacio5** [comment]
Did you see the [debugging tutorial](https://github.com/espressif/vscode-esp-idf-extension/blob/master/docs/tutorial/debugging.md) which explains those configuration steps?

Can you share the output of the `ESP-IDF: Doctor command` and the debug adapter output log (CTRL+Shift+ U -> ESP-IDF Debug Adapter) to see if there is any error there ?

**Comment by mickpf** [comment]
Hello Brian,

1. When I created the issue yesterday and wanted to save it, GitHub had problems saving it. About an hour later, after changing the text locally in a text editor, I was able to save the issue, but later I only saw the old text! I'm sorry but it's not my fault if the issue was created multiple times...

2. Yes, I read all instructions and tutorials available and followed the instructions before contacting you! I had already taken the time for it!

3. I noticed that the OpenOCD does not start automatically, although I switched the mode to "auto". I always have to start it manually. And it cannot be stopped, continues to run and blocks the ports even if I stop it or leave VS Code and start it again.

4. This default settings for launching has no effects. I tried it before I changed the file "launch.json".

5. Hopefully you'l find the output of the "Doctor command" and the "esp_idf_vsc_ext.log" in the attachment... I can't find any log file(s) for the debug adapter!

6. Pressing the key combination <CTRL>+<SHIFT>+U inside VS-Code has no effect! Is it the right combination?

Kind Regards,
Michael
[esp_idf_vsc_ext.log](https://github.com/espressif/vscode-esp-idf-extension/files/6590084/esp_idf_vsc_ext.log)
[Output_Doctor_Command.log](https://github.com/espressif/vscode-esp-idf-extension/files/6590086/Output_Doctor_Command.log)
